client/driver/docker.go

```diff
@@ -686,7 +686,7 @@ func (d *DockerDriver) Open(ctx *ExecContext, handleID string) (DriverHandle, er
 	if !found {
 		return nil, fmt.Errorf("Failed to find container %s: %v", pid.ContainerID, err)
 	}
-	logCollector, pluginClient, err := createLogCollector(pluginConfig, d.config.LogOutput, d.config)
+	exec, pluginClient, err := createExecutor(pluginConfig, d.config.LogOutput, d.config)
```

Comment:
This stuff has to be multi-error otherwise you can leak the executor. 

1) Launch docker + executor
2) Docker engine fails
3) Restart nomad
4) Leak executor